http/file_server.ts

```diff
@@ -643,30 +649,43 @@ async function createServeDirResponse(
   }
 
   const fsPath = join(target, normalizedPath);
-  const fileInfo = await Deno.stat(fsPath);
-
-  // For files, remove the trailing slash from the path.
-  if (fileInfo.isFile && url.pathname.endsWith("/")) {
-    url.pathname = url.pathname.slice(0, -1);
-    return Response.redirect(url, 301);
-  }
-  // For directories, the path must have a trailing slash.
-  if (fileInfo.isDirectory && !url.pathname.endsWith("/")) {
-    // On directory listing pages,
-    // if the current URL's pathname doesn't end with a slash, any
-    // relative URLs in the index file will resolve against the parent
-    // directory, rather than the current directory. To prevent that, we
-    // return a 301 redirect to the URL with a slash.
-    url.pathname += "/";
-    return Response.redirect(url, 301);
-  }
-
-  // if target is file, serve file.
-  if (!fileInfo.isDirectory) {
-    return serveFile(req, fsPath, {
-      etagAlgorithm,
-      fileInfo,
-    });
+  let fileInfo: Deno.FileInfo | undefined;
+  try {
```

Comment:
I think we shouldn't do `try-catch` here. Because we catch the error here, the error logging in [this line](https://github.com/denoland/deno_std/blob/main/http/file_server.ts#L580) is affected (The not found error is not logged when `--verbose` option is given).

There are 3 patterns of returning NotFound HTTP response in `createServeDirResponse` (Search for `return createCommonResponse(Status.NotFound)`). We should replace those 3 with some helper function calls.

I think we can use the helper like the below:

```ts
async function createNotFound(target: string, fallback: boolean) {
  if (fallback) {
    try {
      const path = join(target, "404.html");
      const fileInfo = await Deno.lstat(path);
      if (fileInfo?.isFile) {
        return new Response((await Deno.open(path)).readable, {
          status: Status.NotFound,
        });
      }
    } catch {
      // pass
    }
  }
  return createCommonResponse(Status.NotFound);
}
```

There are also 2 places in `serveFile` where `createCommonResponse(Status.NotFound)` is returned. But those 2 place can be ignored because those code paths are only used when the user use `serveFile` API directly.